src/mono/mono/metadata/assembly.c

```diff
@@ -375,6 +375,10 @@ mono_assemblies_unlock ()
 /* If defined, points to the bundled assembly information */
 static const MonoBundledAssembly **bundles;
 
+#ifdef ENABLE_NETCORE
+static const MonoBundledSatelliteAssembly **satellite_bundles;
```

Comment:
I think you can avoid some of the ifdefs by just having this be defined on non-netcore but not have the registration function actually set it to anything? Then some of the comparisons like `if (bundles != NULL || satellite_bundles != NULL)` should be fine even on legacy.